ui/src/flux/components/SourceDropdown.tsx

```diff
@@ -91,8 +92,13 @@ class SourceDropdown extends PureComponent<Props> {
   }
 
   private get selectedID(): string {
-    const {service} = this.props
-    return service.sourceID + '-' + service.id
+    const {service, source} = this.props
+
+    if (service) {
+      return service.sourceID + '-' + service.id
```

Comment:
Could we use string interpolation here? I think it looks nicer.

Reply:
yeppers